### DISCUSSION TITLE
Support for Stream since rabbitmq 3.9

### DISCUSSION THREAD

**Discussion by hlf2016** [php-amqplib/php-amqplib#998] [Q&A] [answered]
## PHP version 
5.6.40

## Q
How to use this lib consuming  messsages of  stream type?
I haven't seen the examples.
Is it supporting [stream](https://www.rabbitmq.com/streams.html)?

**Comment by Furgas** [comment]
Any news on that? Is there other PHP library supporting streams?

**Comment by lukebakken** [comment, marked as answer]
You can consume from streams as though they are AMQP queues:

https://www.rabbitmq.com/streams.html#usage

https://www.rabbitmq.com/stream-core-plugin-comparison.html

**Comment by Furgas** [comment]
Ok, you're right. Thanks.

**Reply by gvillela7** [reply]
Hi,
Were you able to consume a queue of type stream?
Can you help me? How did you do?
I'm trying like this:
`
$this->channel->basic_consume(
            'stream_queue',
            '',
            false,
            false,
            false,
            false,
            [$this, 'callback'],
            null,
            ['x-stream-offset', 'first'],
        );
`

**Reply by mgraichy** [reply]
I'm about a year and a couple of months late on this, but here's how I got my streams working using the AMQP:
```
$streamChannel->queue_declare(
    queue:       'stream_connection',
    durable:     true,
    auto_delete: false,
    exclusive:   false,
    arguments:   ['x-queue-type' => ['S', 'stream']]
);

$streamChannel->basic_qos(
    prefetch_count: 100,
    prefetch_size: null,
    a_global: false
);

$streamChannel->basic_consume(
    queue:    'stream_connection',
    callback: 'cb',
    // Same as not specifying any offset:
    arguments: ['x-stream-offset' => ['S', 'next']]
);
```

'S' is for strings (`AMQPAbstractCollection.php` line 82)

**Comment by reggin** [comment]
Since rabbitmq 3.13 was added stream filtering. Can it be done now by this package?
And do you have plans to add stream as separate classes for using?
